src/Northwind.Web/Controllers/Api/ProductsController.cs

```diff
@@ -0,0 +1,158 @@
+                var productModel = _mapper.Map<ProductModel>(product);
+                return Ok(productModel);
+            }
+            catch (NotFoundException)
+            {
+                return NotFound();
+            }
+            catch
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, "Failed to get the product");
+            }
+        }
+
+        /// <summary>
+        /// Create a concrete Product
+        /// </summary>
+        /// <returns>A create Product</returns>
+        /// <response code="201">Returns new item's values</response>
+        /// <response code="400">If the validations failed</response>    
+        /// <response code="500">An error occurred on the server side</response>  
+        [HttpPost]
+        public async Task<ActionResult<CreateProductModel>> Post(CreateProductModel model)
+        {
+            try
+            {
+                if (ModelState.IsValid)
+                {
+                    var product = _mapper.Map<ProductEntity>(model);
+                    await _productService.CreateAsync(product);
+                    return Created($"/api/products/{product.ProductID}", model);
+                }
+
+                return BadRequest();
+            }
+            catch
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, "Failed to create the product");
+            }
+        }
+
+        /// <summary>
+        /// Update a Product
+        /// </summary>
+        /// <returns>An updated Product</returns>
+        /// <response code="400">If the validations failed</response>    
+        /// <response code="404">If the item is null or was not found</response>   
+        /// <response code="500">An error occurred on the server side</response>  
+        [HttpPut]
+        public async Task<ActionResult<UpdateProductModel>> Put(UpdateProductModel model)
+        {
+            try
+            {
+                if (ModelState.IsValid && model.ProductID != 0)
+                {
+                    var product = _mapper.Map<ProductEntity>(model);
+                    await _productService.UpdateAsync(product);
+                    return model;
+                }
+
+                return BadRequest();
+            }
+            catch (NotFoundException)
+            {
+                return NotFound();
+            }
+            catch
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, "Failed to update the product");
+            }
+        }
+
+        /// <summary>
+        /// Delete a concrete Product
+        /// </summary>
+        /// <returns>A deleted Product</returns>
+        /// <response code="404">If the item is null or was not found</response>    
+        /// <response code="500">An error occurred on the server side</response>  
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> Delete(int id)

```

Comment:
Checking for valid values will make it possible to avoid unnecessary request to DB